### PR TITLE
Using plain old requirements.txt and removing Pipfile

### PR DESCRIPTION
**Pull request by AhmedSoliman**
- Stopping the use of Pipenv since it's not really ready for library use.
- Use the standard requirements.txt approach
- Cleaning up documentation
- Adjusting setup.py to support nosetests directly
- Adjusting travis config file accordingly

### CODE REVIEW

**Comment by ylectric** [comment]
Happy to review this later today.

File: README.md [hunk 1]
```diff
@@ -73,27 +73,36 @@ pip3 install python-nubia
...
-```bash
-pip3 install pipenv
-```
 You can either setup.py to build a tarball, or use pipenv to setup a virtualenv with all the dependencies installed.
 
 ## Running example in virtualenv:
 
+It's often best to create a virtualenv to contain the dependencies required for python-nubia project.
+```bash
+ python3 -m virtualenv .venv
```
Comment (ylectric) [inline README.md]: Please remove the leading whitespace.
Also, shouldn't it be `venv` module? I get `No module named virtualenv` error when following these instructions.

Comment (ylectric) [reply README.md]: Checked this on Python 3.6 an 3.7, it should be `venv`.

Comment (AhmedSoliman) [reply README.md]: Fixed.

**Pull request closed without merging**